tests/test_nested_suites.c

```diff
@@ -0,0 +1,157 @@
+    return MUNIT_OK;
+}
+
+
+static MunitResult test_func_2(const MunitParameter params[], void* data) {
+    ((test_call_t*) data)->test_calls_b += 1;
+    return MUNIT_OK;
+}
+
+static MunitTest test_a[] = {
+        {"test_a1", test_func_1, test_setup, test_teardown, MUNIT_TEST_OPTION_NONE, NULL},
+        {"test_a2", test_func_2, test_setup, NULL, MUNIT_TEST_OPTION_NONE, NULL},
+        {NULL, NULL, NULL, NULL, 0, MUNIT_TEST_OPTION_NONE}
+};
+
+
+static MunitTest test_b[] = {
+        {"test_b1", test_func_1, NULL, test_teardown, MUNIT_TEST_OPTION_NONE, NULL},
+        {"test_b2", test_func_2, NULL, NULL, MUNIT_TEST_OPTION_NONE, NULL},
+        {NULL, NULL, NULL, NULL, 0, MUNIT_TEST_OPTION_NONE}
+};
+
+static MunitSuite main_suite[] = {
+        {"nested_a/", test_a, suite_setup, suite_teardown, NULL, 7, MUNIT_SUITE_OPTION_NONE},
+        {"nested_b/", test_b, NULL, NULL, NULL, 5, MUNIT_SUITE_OPTION_NONE},
+        {NULL, NULL, NULL, NULL, NULL, 0, MUNIT_SUITE_OPTION_NONE}
+};
+
+static const MunitSuite root_suite = {
+        "",
+        /* No tests running for root suite needed */
+        NULL,
+        NULL,
+        NULL,
+        main_suite,
+        0,
+        MUNIT_SUITE_OPTION_NONE
+};
+
+
+int main(int argc, char *argv[MUNIT_ARRAY_PARAM(argc + 1)]) {
+    //test all suites run
+    char* args_all_tests[] = {
+            "Program",
+            "--no-fork",
+            NULL
+    };
+    char* args_test_a[] = {
+            "Program",
+            "--no-fork",
+            "nested_a",
+            NULL
+    };
+    char* args_test_b[] = {
+            "Program",
+            "--no-fork",
+            "nested_b",
+            NULL
+    };
+    char* args_test_a1[] = {
+            "Program",
+            "--no-fork",
+            "nested_a/test_a1",
+            NULL
+    };
+    char* args_test_b2[] = {
+            "Program",
+            "--no-fork",
+            "nested_b/test_b2",
+            NULL
+    };
+    char* args_no_test_case[] = {
+            "Program",
+            "--no-fork",
+            "something",
+            NULL
+    };
+
+    //Test calling cases by name and suite name
+    test_call_t calls;
+    reset_test(&calls);
+    munit_assert(munit_suite_main(&root_suite, (void *) &calls, 2, args_all_tests) == EXIT_SUCCESS);
+    assert_calls(&calls, 1, 1, 14, 12, 12, 12);
+
+    reset_test(&calls);
+    munit_assert(munit_suite_main(&root_suite, (void *) &calls, 3, args_test_a) == EXIT_SUCCESS);
+    assert_calls(&calls, 1, 1, 14, 7, 7, 7);
+
+    reset_test(&calls);
+    munit_assert(munit_suite_main(&root_suite, (void *) &calls, 3, args_test_b) == EXIT_SUCCESS);
+    assert_calls(&calls, 0, 0, 0, 5, 5, 5);
+
+    reset_test(&calls);
+    munit_assert(munit_suite_main(&root_suite, (void *) &calls, 3, args_test_a1) == EXIT_SUCCESS);
+    assert_calls(&calls, 1, 1, 7, 7, 7, 0);
+
+    reset_test(&calls);
+    munit_assert(munit_suite_main(&root_suite, (void *) &calls, 3, args_test_b2) == EXIT_SUCCESS);
```

Comment:
it was not clear for me why b2 executes 0..0, 5,
while b is 0,0,0,5,5,5
So I introduced two comments (lines 117, 133)

Reply:
Suite b does not run suite setup or teardown, that is the first two 0, 0
test_b does not run any test setup or teardown, that is the next two 0,0 
test_b runs 5 times over test_func_2, so test_func_1 is not executed. There is the next 0, 5

Reply:
the all_suite_b test, however, runs 0,0,0,0,0,5 from the example above and 0,0,0,5,5,0 from the nested_b/test_b2 case.
Sum them up and you have 0,0,0,5,5,5